variant-inc/actions-setup
# [CLOUD-1854] defaulting AWS_REGION and AWS_DEFAULT_REGION if not passed in

# Description

Fixes [CLOUD-1854](https://drivevariant.atlassian.net/browse/CLOUD-1854)

I will retag v1 once the PR is approved.

## Type of change


## How Has This Been Tested?

[test branch](https://github.com/variant-inc/dx-demo-cron/tree/t/test-aws-region)

Test Case 1 [AWS_REGION & AWS_DEFAULT_REGION unset]:
- Steps
  - set `actions/setup` to be `uses: variant-inc/actions-setup@f/CLOUD-1854`
  - remove any environment declaration for AWS_DEFAULT_REGION and AWS_DEFAULT_REGION
  - commit and push
- Result
 - build should complete. Image should be in the variant-prod account ECR, region us-east-1
 - https://github.com/variant-inc/dx-demo-cron/runs/7153270443?check_suite_focus=true
 - https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/dx-demo-cron/deployments/releases/test-0.87.1

Test Case 2 [AWS_REGION=us-east-2 & AWS_DEFAULT_REGION=us-east-2]:
- Steps
  - set `actions/setup` to be `uses: variant-inc/actions-setup@f/CLOUD-1854`
  - set both variables in the workflow
  ```
  env:
  AWS_REGION: us-east-2
  AWS_DEFAULT_REGION: us-east-2
  ```
  - commit and push
- Result
 - build should complete. Image should be in the variant-prod account ECR, region us-east-2
 - https://github.com/variant-inc/dx-demo-cron/runs/7153254325?check_suite_focus=true
 - https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/dx-demo-cron/deployments/releases/test-0.86.1

Test Case 3 [AWS_REGION unset & AWS_DEFAULT_REGION=us-east-1]:
- Steps
  - set `actions/setup` to be `uses: variant-inc/actions-setup@f/CLOUD-1854`
  - set AWS_DEFAULT_REGION=us-east-1 in the flow but remove any declaration for AWS_REGION
  ```
  env:
  AWS_DEFAULT_REGION: us-east-1
  ```
  - commit and push
- Result
 - build should complete. Image should be in the variant-prod account ECR, region us-east-1
 - https://github.com/variant-inc/dx-demo-cron/runs/7153276698?check_suite_focus=true
 - https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/dx-demo-cron/deployments/releases/test-0.88.1

*NOTE* setting AWS_DEFAULT_REGION to a value other than us-east-1 will result in an error as a result of bug [CLOUD-1862](https://drivevariant.atlassian.net/browse/CLOUD-1862)


## Checklist
